src/internal_rep.rs

```diff
@@ -866,6 +866,7 @@ impl<'a> ClassList<'a> {
         permission: &CascadeString,
         context: &BlockContext<'_>,
         file: &SimpleFile<String, String>,
+        original_class: Option<&CascadeString>,
```

Comment:
Not overly wild about adding this aspect, as it seems like it exposes an internal implementation detail to callers unnecessarily.

Without having thought about it too much, would it work to check for the "2" versions in the error message and just convert back based on the internal mapping between originals and "2"s?  Or do we need to know that we're in the recursive lookup at that point?

Reply:
It depends on how we want to present the error to the user.  With this solution we will tell them non-2 and 2 correctly, which we need to know if we have entered the function recursively to check the 2s.

If we just always want to tell them the non 2s I can just simply map it back.

Reply:
From a cil, AVC, non-cascade standpoint I think it's better to tell them if it is a non-2 or a 2, but since we just combine them in cascade it may be okay to just map it back.

Reply:
I think my preference here is to create a new verify_permission_helper() to pass the recursive parameter to, rather than exposing it to callers.